lusidtools/cocoon/cocoon_printer.py

```diff
@@ -0,0 +1,171 @@
+import pandas as pd
+import numpy as np
+
+
+def format_instruments_response(response: dict, file_type: str = "instruments") -> (
+pd.DataFrame, pd.DataFrame, pd.DataFrame):
+    """
+    This function unpacks a response from instrument requests and returns successful, failed and errored statuses for
+    request constituents.
+    :param dict response: response from Lusid-python-tools
+    :param str file_type: file_type of Lusid-python-tools
+    :return: pd.DataFrame: Successful calls from request
+    :return: pd.DataFrame: Error responses from request that fail (APIExceptions: 400 errors)
+    :return: pd.DataFrame: Failed responses that LUSID rejected
+    """
+    items_success, items_failed, items_error = [], [], []
+    if "success" not in response[file_type].keys() or \
+            "errors" not in response[file_type].keys():
+        raise ValueError("Didn't recognise response shape. Must include 'errors', and 'success' keys")
+
+    # get successes and failures
+    for batch in response[file_type]["success"]:
+        [items_success.append(batch.values[key].name) for key in batch.values.keys()]
+        if len(response[file_type]["success"][0].failed) > 0:
+            [items_failed.append(batch.failed[key].name) for key in batch.failed.keys()]
+        else:
```

Comment:
Can you just define them both inline here using list comprehension instead of initialising them above?

Reply:
```items_success = [batch.values[key].quote_id.quote_series_id.instrument_id for batch in
                     response[file_type]["success"] for key in batch.values.keys()]

    if len(response[file_type]["success"][0].failed) > 0:
        items_failed = [batch.failed[key].quote_id.quote_series_id.instrument_id for batch in
                        response[file_type]["success"] for key in batch.failed.keys()]```

I have not included the if statement inline with the comprehension because the line gets too long and unreadable